data/campaigns/Heir_To_The_Throne/scenarios/04_The_Bay_of_Pearls.cfg

```diff
@@ -1,4 +1,77 @@
 #textdomain wesnoth-httt
+
+#define ADVISORS
+    "Elvish Sylph," +
+    "Great Mage," +
+
+    "Elvish Marshal,Elvish Champion," +
+    "Elvish Avenger,Elvish Sharpshooter," +
+    "Elvish Outrider," +
+    "Elvish Shyde,Elvish Enchantress," +
+    "Paladin,Grand Knight," +
+    "Mage of Light,Arch Mage,Silver Mage," +
+
+    "Elvish Captain,Elvish Hero," +
+    "Elvish Ranger,Elvish Marksman," +
+    "Elvish Rider," +
+    "Elvish Druid,Elvish Sorceress," +
+    "Knight,Lancer," +
+    "White Mage,Red Mage," +
+
+    "Elvish Fighter," +
+    "Elvish Archer," +
+    "Elvish Scout," +
+    "Elvish Shaman," +
+    "Horseman," +
+    "Mage"
+#enddef
+
+#define MERMEN
+    "Merman Triton,Merman Hoplite," +
+    "Merman Javelineer,Merman Entangler," +
+    "Mermaid Diviner,Mermaid Siren," +
+    
+    "Merman Warrior," +
+    "Merman Spearman,Merman Netcaster," +
+    "Mermaid Priestess,Mermaid Enchantress," +
+    
+    "Merman Fighter," +
+    "Merman Hunter," +
+    "Mermaid Initiate"
+#enddef
+
+#define UnitSpeaks RoleName UnitList SpokenLine
```

Comment:
This macro seems to utterly defeat the point of using `[role]` in the first place. If I'm not mistaken, you could simply delete lines 53-56 and it would still work. Furthermore, in that case the `[if]` conditional is meaningless since a `[message]` is automatically and silently skipped if no speaker is found.

More generally, I'm not convinced this is the best solution. Did you consider adding a death event to reassign essential roles? Something like the following:

```wml
[event]
  name=die
  [filter]
    role=some_role_name
  [/filter]
  [role]
    role=some_role_name
    side=1
    type=whatever
  [/role]
[/event]
```

I guess you'd need two of them, one for the thankful merfolk and one for the advisor.